deployment/ccip/changeset/v1_6/cs_home_chain.go

```diff
@@ -667,3 +690,316 @@ func removeNodesLogic(env deployment.Environment, c RemoveNodesConfig) (deployme
+	nopsIDByName := make(map[string]uint32)
+	for _, nop := range c.ExistingNops {
+		id := nodeOperatorIDByNop(nopsByID, nop)
+		if id == 0 {
+			return deployment.ChangesetOutput{}, fmt.Errorf("node operator %s does not exist in cap reg %s", nop.Name, homeChainState.CapabilityRegistry.Address().String())
+		}
+		nopsIDByName[nop.Name] = id
+	}
+	// check if the number of node operators to update is same as the number of existing node operators found in the CapabilitiesRegistry contract
+	if len(nopsIDByName) != len(c.NopUpdates) {
+		return deployment.ChangesetOutput{}, errors.New("number of existing node operators found in cap reg and node operators to update must be same")
+	}
+	nops := make([]capabilities_registry.CapabilitiesRegistryNodeOperator, 0, len(c.NopUpdates))
+	nopIDs := make([]uint32, 0, len(c.NopUpdates))
+	for name, nop := range c.NopUpdates {
+		id, ok := nopsIDByName[name]
+		if !ok {
+			return deployment.ChangesetOutput{}, fmt.Errorf("node operator %s does not exist in cap reg %s", name, homeChainState.CapabilityRegistry.Address().String())
+		}
+		nops = append(nops, nop)
+		nopIDs = append(nopIDs, id)
+		// validate that the sender is the owner of the CapabilitiesRegistry or the admin of the existing node operator
+		existingNop := nopsByID[id]
+		err = validateAccess(env.GetContext(), homeChain, homeChainState, homeChainState.CapabilityRegistry, existingNop, c.MCMSConfig != nil, false)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to validate access for node operator %s: %w", nop.Name, err)
+		}
+	}
+	deployerGroup := changeset.NewDeployerGroup(env, state, c.MCMSConfig).
+		WithDeploymentContext("update nops in cap reg")
+	txOpts, err := deployerGroup.GetDeployer(homeChain.Selector)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to get deployer for home chain %d: %w", homeChain.Selector, err)
+	}
+	_, err = homeChainState.CapabilityRegistry.UpdateNodeOperators(txOpts, nopIDs, nops)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to create transaction to update node operators in cap reg %s: %w", homeChainState.CapabilityRegistry.Address().String(), err)
+	}
+	return deployerGroup.Enact()
+}
+
+// validateAccess checks -
+//
+//		if the sender is the owner of the CapabilitiesRegistry or the admin of the node operator, if onlyOwner is false
+//	    if the sender is the owner of the CapabilitiesRegistry, if onlyOwner is true
+func validateAccess(
+	ctx context.Context,
+	chain deployment.Chain,
+	chainState changeset.CCIPChainState,
+	capReg *capabilities_registry.CapabilitiesRegistry,
+	nop capabilities_registry.CapabilitiesRegistryNodeOperator,
+	isMCMS bool,
+	onlyOwner bool,
+) error {
+	owner, err := capReg.Owner(&bind.CallOpts{
+		Context: ctx,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get owner of CapabilitiesRegistry %s: %w", capReg.Address().String(), err)
+	}
+	sender := chain.DeployerKey.From
```

Comment:
nit: Maybe this is something we can add to the deployer group to know which address will ownership will be validated against